### PR TITLE
[CLI-788] Update CLI version in muckrake during release

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Discussed with the quality eng team, and agreed we should transition to testing the CLI against CP in nightly muckrake runs. They no longer have the capacity to keep .x branches stable, so our tests will be run nightly on the branch builder tool. This matches the testing strategy of our Confluent Cloud tests, although those run more frequently (every hour).

Example PR: https://github.com/confluentinc/muckrake/pull/2081

Test & Review
-------------
Ran with `DRY_RUN=true`.

### CODE REVIEW

File: mk-files/release.mk [hunk 1]
```diff
@@ -148,3 +150,32 @@ download-licenses:
...
+.PHONY: update-muckrake
+update-muckrake:
+	$(eval DIR=$(shell mktemp -d))
+	$(eval CLI_RELEASE=$(DIR)/cli-release)
+	$(eval MUCKRAKE=$(DIR)/muckrake)
+
+	git clone git@github.com:confluentinc/cli-release.git $(CLI_RELEASE) && \
+	cd $(CLI_RELEASE) && \
+	version=$$(ls release-notes | $(SED) "s/.json$$//" | sort --version-sort | tail -1) && \
+	git clone git@github.com:confluentinc/muckrake.git $(MUCKRAKE) && \
```
Comment (MuweiHe) [inline mk-files/release.mk:163]: muckrake is being cloned under cli-release?

Comment (brianstrauch) [reply mk-files/release.mk:163]: Yup, we need to clone it before we can open a PR.

Comment (MuweiHe) [reply mk-files/release.mk:163]: i'm confused it looks like muckrake is being clones inside of cli-release, but from `$(eval CLI_RELEASE=$(DIR)/cli-release) $(eval MUCKRAKE=$(DIR)/muckrake)` they're at the same level of dir

Comment (brianstrauch) [reply mk-files/release.mk:163]: The second argument to `git clone` is the directory that we're cloning into!

Comment (MuweiHe) [reply mk-files/release.mk:163]: oh haha

**Review by MuweiHe — APPROVED** [review]
thanks!

**Pull request merged**